src/encoded/static/components/visualizations.js

```diff
@@ -244,13 +246,273 @@ export class TestViz extends React.Component {
+            this.setState({
+                currentTarget: [...this.state.currentTarget, `table${clickID}`],
+            });
+        } else {
+            const removedTarget = this.state.currentTarget.filter(element => element !== `table${clickID}`);
+            this.setState({
+                currentTarget: removedTarget,
+            });
+        }
+    }
+
+    handleSearch(event) {
+        // Unsanitized search term entered by user for display
+        this.setState({ unsanitizedSearchTerm: event.target.value });
+        // Search term entered by the user
+        const filterVal = String(sanitizedString(event.target.value));
+        this.setState({ searchTerm: filterVal });
+    }
+
+    clearSearch() {
+        // clear both search terms
+        this.setState({
+            unsanitizedSearchTerm: '',
+            searchTerm: '',
+        });
+    }
+
+    expandTerms() {
+        const allTargets = [];
+        for (let idx = 0; idx <= Object.keys(this.state.chartData).length; idx += 1) {
+            allTargets.push(`table${idx}`);
+        }
+        this.setState({
+            currentTarget: allTargets,
+        });
+    }
+
+    collapseTerms() {
+        this.setState({
+            currentTarget: [],
+        });
+    }
+
+    drawCharts() {
+        const allData = this.props.context['@graph'];
+        let data = allData;
+        if (this.props.dataFilter === 'chip') {
+            data = allData.filter(d => d.assay_title === 'ChIP-seq');
+        } else if (this.props.dataFilter === 'qtl') {
+            data = allData.filter(d => (d.annotation_type && d.annotation_type.indexOf('QTL') !== -1));
+        } else if (this.props.dataFilter === 'dnase') {
+            data = allData.filter(d => (d.assay_title === 'FAIRE-seq' || d.assay_title === 'DNase-seq'));
+        } else if (this.props.dataFilter === 'chromatin') {
+            data = allData.filter(d => (d.annotation_type === 'chromatin state'));
+        }
+        const fakeFacets = [];
+        data.forEach((d) => {
+            if (fakeFacets[d.biosample_ontology.term_name]) {
+                fakeFacets[d.biosample_ontology.term_name] += 1;
+            } else {
+                fakeFacets[d.biosample_ontology.term_name] = 1;
+            }
+        });
+        // sort the fake facets
+        const keys = Object.keys(fakeFacets);
+        keys.sort((a, b) => (fakeFacets[b] - fakeFacets[a]));
+        const sortedFakeFacets = [];
+        keys.forEach((key) => {
+            sortedFakeFacets[key] = fakeFacets[key];
+        });
+        const chartData = sortedFakeFacets;
+        const chartTitle = this.props.displayTitle;
+        const chartArray = Object.keys(chartData).map(key => chartData[key]);
+        const chartMax = Math.max(...chartArray);
+        // compute left margin
+        let leftMargin = 60;
+        Object.keys(chartData).forEach((d) => {
+            leftMargin = Math.max(d.length * 7, leftMargin);
```

Comment:
This looks like it just gets the last chartData array length or leftMargin.

Reply:
I think I need this loop because I have to find the biosample name with the most characters and that determines what the left margin should be.

Reply:
I see that now. nvm.